pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -344,19 +302,17 @@ func (r *ReconcileOperConfig) Reconcile(ctx context.Context, request reconcile.R
 	}
 	objs = append([]*uns.Unstructured{app}, objs...)
 
-	// Set up the Pod reconciler before we start creating DaemonSets/Deployments/StatefulSets
-	daemonSets := []statusmanager.ClusteredName{}
-	deployments := []statusmanager.ClusteredName{}
-	statefulSets := []statusmanager.ClusteredName{}
 	relatedObjects := []configv1.ObjectReference{}
 	relatedClusterObjects := []network.RelatedObject{}
 	for _, obj := range objs {
-		if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "DaemonSet" {
-			daemonSets = append(daemonSets, statusmanager.ClusteredName{ClusterName: apply.GetClusterName(obj), Namespace: obj.GetNamespace(), Name: obj.GetName()})
-		} else if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "Deployment" {
-			deployments = append(deployments, statusmanager.ClusteredName{ClusterName: apply.GetClusterName(obj), Namespace: obj.GetNamespace(), Name: obj.GetName()})
-		} else if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "StatefulSet" {
-			statefulSets = append(statefulSets, statusmanager.ClusteredName{ClusterName: apply.GetClusterName(obj), Namespace: obj.GetNamespace(), Name: obj.GetName()})
+		// Label all DaemonSets, Deployments, and StatefulSets with the label that generates Status.
```

Comment:
Do we ever need to remove the label, or all outdated object should be deleted by deleteRelatedObjectsNotRendered?

Reply:
I doubt we will ever remove the label, but doing so would be handled correctly: the informer would trigger and we would re-generate status from the remaining objects.

Reply:
(The informer treats removing the label as a "delete")

Reply:
I mean do we need to clean the label for non-already-related objects - in this PR we only add this label, but if some object that was labeled gets outdated (e.g during upgrade old daemonsets will still be labeled) then controller will keep considering that object as a part of a status since it has the label.
So the question is - is it even possible, or all "stale" objects should be just deleted fast enough to not confuse status manager?

Reply:
I'm not sure I entirely understand the scenario you're talking about. But, in any case, I don't think the status manager will be "confused". It will always generate a reasonable status based on the objects in the cluster with the label. If a stale object is deleted, then it's deletion will cause status to be re-generated.

We're no longer maintaining a list of desired objects. The source of truth is the informer's store. So now, hopefully, we should be *more consistent*.